packages/core/src/queries/passcode.ts

```diff
@@ -11,24 +11,45 @@ import { DeletionError } from '#src/errors/SlonikError/index.js';
 
 const { table, fields } = convertToIdentifiers(Passcodes);
 
+type FindByIdentifierAndTypeProperties = {
+  type: VerificationCodeType;
+  email?: string;
+  phone?: string;
+};
+
+const buildSqlForFindByJtiAndType = (jti: string, type: VerificationCodeType) => sql`
+  select ${sql.join(Object.values(fields), sql`, `)}
+  from ${table}
+  where ${fields.interactionJti}=${jti} and ${fields.type}=${type} and ${fields.consumed} = false
+`;
+
+const buildSqlForFindByIdentifierAndType = ({
+  email,
+  phone,
+  type,
+}: FindByIdentifierAndTypeProperties) => sql`
+  select ${sql.join(Object.values(fields), sql`, `)}
+  from ${table}
+  where 
+    ${conditionalSql(email, (email) => sql`${fields.email}=${email}`)}
+    ${conditionalSql(phone, (phone) => sql`${fields.phone}=${phone}`)}
+    and ${fields.type}=${type} and ${fields.consumed} = false
```

Comment:
if both `phone` and `email` are empty, this `and` will cause a sql error. Thought `email` and `phone` will not be empty at the same time in our current business.

Reply:
They will not be empty at the same time, as we have guarded the payload at API route level. But you are also right because at this query level we don't have a guard for that and it might cause potential issues in future development. I'll try to improve this part.

Reply:
updated

Reply:
maybe add a comment here to explain that, per our current business, the email and phone always have one of them empty and another with valid value? so that others can figure out the logic easier.

Reply:
> maybe add a comment here to explain that, per our current business, the email and phone always have one of them empty and another with valid value? so that others can figure out the logic easier.

Sure, no problem